lib/helpers.js

```diff
@@ -1,69 +1,97 @@
-      if (!user) return res.status(401).json({errors: [{message: 'Invalid Authorization token.'}]});
-      if (user.error) return next(user.error);
-      req.user = user.toJSON();
-      if (payload.driver_id) {
-        const driver = await new Driver({id: payload.driver_id}).fetch();
-        if (driver) {
-          req.driver = driver.toJSON();
-          req.driver.blacklist = await driver.blacklist();
-        }
-      }
-      req.user.token = token;
-      next();
-    } catch (e) {
-      console.error('Invalid authorization header. ', e.message);
-      res.status(401).json({ errors: [{message: 'Invalid authorization header'}]});
+  await fn()
+}
+
+async function assignDriverIfRequired (req) {
+  if (req.payload.driver_id != null) {
+    const driver = await new Driver({ id: req.payload.driver_id }).fetch()
+    if (driver != null) {
+      req.driver = driver.toJSON()
+      req.driver.blacklist = await driver.blacklist()
     }
-  },
-  requireResetPasswordToken: async function requireResetPasswordToken(req, res, next) {
-    try {
-      let token = req.query.token || req.body.token
-      if (token) {
-        req.errors = []
-        var payload = authToken.decode(token);
-        let user = await new User({id: payload.id}).fetch();
-        if (user && Date.now() <= payload.expiration_date){
-          req.user = user.toJSON();
-          req.user.token = token;
-        }
-        else
-          req.user = null;
-      } else {
-        req.errors = ['Token is necesary'];
-      }
-      next();
-    } catch (e) {
-      console.error('Invalid Token. ', e.message);
-      req.errors = [...(req.errors || []), 'Invalid Token'];
-      next();
+  }
+}
+
+async function requireAuthentication (req, res, next) {
+  await withErrorHandler(res, async () => {
+    verifyAndDecodeTokenPayload(req, extractToken(req))
+    await withUser(req, res, next, async () => {
+      await assignDriverIfRequired(req)
+      next()
+    })
+  })
+}
+
+async function requireAdminAuthentication (req, res, next) {
+  await withErrorHandler(res, () => {
+    const token = extractToken(req)
+    if (safeEquals(process.env.ADMIN_TOKEN, token)) {
+      req.token = token
+      return next()
+    } else {
+      return res.status(401).json(genericErrorMessage('Invalid Authorization token.'))
     }
-  },
-  requireAdminAuthentication: function requireAdminAuthentication(req, res, next) {
-    try {
-      let header = req.header('Authorization');
-      let token = header.split(' ')[1];
+  })
+}
 
-      if (process.env.ADMIN_TOKEN == token) {
-        req.token = token;
-        return next();
-      } else {
-        return res.status(401).json({errors: [{message: 'Invalid Authorization token.'}]});
-      }
-    } catch (e) {
-      console.error('Invalid authorization header. ', e.message);
-      res.status(401).json({ errors: [{message: 'Invalid authorization header'}]});
+async function requireResetPasswordToken (req, _, next) {
```

Comment:
I think is better to keep the name "res" instead of change it to underscore even if we don't use it because it will cause confusion for the beginners.

Reply:
ok ok, will change